quiz.py

```diff
@@ -0,0 +1,148 @@
+import random
+from abc import ABCMeta, abstractmethod
+from sqlalchemy.sql.expression import exists
+from sqlalchemy.sql.functions import func
+from models.painter import Painter
+from models.painting import Painting
+from models.saint import Saint
+
+
+class MultipleChoiceQuestion:
+    def __init__(self, element, correct_option, option_list):
+        self._element = element
+        self.element = element.values()[0]
+        self.correct_option = correct_option
+        self.option_list = option_list
+
+
+class Quiz:
+    __metaclass__ = ABCMeta
+
+    def __init__(self):
+        self.number_of_rounds = 0
+        self.score = []
+
+        # current
+        self.current_question = None
+
+        self.right_guesses = sum(self.score)
+        self.wrong_guesses = len(self.score) - self.right_guesses
+
+    def get_score(self):
+        return self.score
+
+    @abstractmethod
+    def generate_question(self):
+        return None
+
+    def update_score(self, correct):
+        if correct:
+            self.right_guesses += 1
+        else:
+            self.wrong_guesses += 1
+        self.score.append(correct)
+
+    def next_question(self):
+        self.number_of_rounds += 1
+        question = self.generate_question()
+        return question
+
+
+class ImageQuiz(Quiz):
+    def __init__(self, num_elements_quiz=4):
+        super(ImageQuiz, self).__init__()
+
+        elements_dict, options_dict = self.initialize_image_quiz(num_elements=10)
+        self.elements_dict = elements_dict
+        self.options_dict = options_dict
+        self.num_elements_quiz = num_elements_quiz
+
+        # Auxiliary attributes
+        self.seen_elements_dict = dict()
+        self.current_question = None
+
+    def initialize_image_quiz(self, num_elements):
```

Comment:
why it is not marked as abstract method (with empty return), since you cant provide a default implementation here?